plugins/armorblox/plugin.spec.yaml

```
@@ -0,0 +1,206 @@
+      title: Detection tag name
+      description: Detection tag name
+      type: string
+      required: false    
+  engagement:
+    fwd_mail_count:
+      title: Forwarded Mail Count
+      description: Forwarded Mail Count
+      type: string
+      required: false
+    reply_mail_count:
+      title: Reply Mail Count
+      description: Reply Mail Count
+      type: string
+      required: false
+  incident:
+    priority:
+      title: Priority
+      description: Priority of the incident
+      type: string
+      required: false
+    tagged:
+      title: Is email tagged
+      description: Is email tagged
+      type: boolean
+      required: false
+    date:
+      title: Incident Date
+      description: Date of the incident
+      type: date
+      required: false
+    policy_names:
+      title: policy_names
+      description: List of policies
+      type: '[]string'
+      required: false
+    title:
+      title: Subject
+      description: Mail subject
+      type: 'string'
+      required: false          
+    resolution_state:
+      title: resolution_state
+      description: Incident resolution state
+      type: string
+      required: false
+    object_type:
+      title: Object Type
+      description: Type of the object
+      type: string
+      required: false 
+    id:
+      title: Incident ID
+      description: Incident ID
+      type: string
+      required: false
+    app_name:
+      title: App Name
+      description: App Name
+      type: string
+      required: false 
+    external_senders:
+      title: External senders
+      description: List of external senders
+      type: "[]string"
+      required: 
+    folder_categories:
+      title: Folder categories
+      description: Folder categories
+      type: "[]string"
+      required: false
+    scl_score:
+      title: SCL Score
+      description: 
+      type: integer
+      required: false
+    incident_type:
+      title: Incident Type
+      description: Incident Type
+      type: string
+      required: 
+    engagements:
+      title: Engagements
+      description: Engagements 
+      type: engagement
+      required: false
+    remediation_actions:
+      title: Remediation Action
+      description: Remediation Action
+      type: "[]string"
+      required: false
+    users:
+      title: users
+      description: List of users
+      type: "[]user"
+      required: false
+    external_users:
+      title: External users
+      description: List of external users
+      type: "[]user"
+      required: false
+    final_detection_tags:
+      title: Detection tags
+      description: Detection tags 
+      type: "[]final_detection_tag"
+      required: false
+
+connection:
+  tenant_name:
+    title: Tenant Name
+    description: Armorblox Tenant Name
+    type: string
+    required: true
+    example: my-tenant-name
+  api_key:
+    type: credential_secret_key
+    title: Armorblox API Key
+    description: Armorblox API Key
```

This is purely optional, the functionality won't change here. It's just good practice to have the pattern of the fields stay consistent.
```suggestion
    title: Armorblox API Key
    description: Armorblox API Key
    type: credential_secret_key
```